ple/games/pong.py

```diff
@@ -42,40 +42,28 @@ def __init__(self, radius, speed, rng,
         self.rect = self.image.get_rect()
         self.rect.center = pos_init
 
-    def line_intersection(self, p0_x, p0_y, p1_x, p1_y, p2_x, p2_y, p3_x, p3_y):
-
-        s1_x = p1_x - p0_x
-        s1_y = p1_y - p0_y
-        s2_x = p3_x - p2_x
-        s2_y = p3_y - p2_y
-
-        s = (-s1_y * (p0_x - p2_x) + s1_x * (p0_y - p2_y)) / (-s2_x * s1_y + s1_x * s2_y)
-        t = (s2_x * (p0_y - p2_y) - s2_y * (p0_x - p2_x)) / (-s2_x * s1_y + s1_x * s2_y)
-
-        return (s >= 0 and s <= 1 and t >= 0 and t <= 1)
-
     def update(self, agentPlayer, cpuPlayer, dt):
 
         self.pos.x += self.vel.x * dt
         self.pos.y += self.vel.y * dt
 
         is_pad_hit = False
 
-        if self.pos.x <= agentPlayer.pos.x + agentPlayer.rect_width:
-            if self.line_intersection(self.pos_before.x, self.pos_before.y, self.pos.x, self.pos.y, agentPlayer.pos.x + agentPlayer.rect_width / 2, agentPlayer.pos.y - agentPlayer.rect_height / 2, agentPlayer.pos.x + agentPlayer.rect_width / 2, agentPlayer.pos.y + agentPlayer.rect_height / 2):
-                self.pos.x = max(0, self.pos.x)
-                self.vel.x = -1 * (self.vel.x + self.speed * 0.05)
-                self.vel.y += agentPlayer.vel.y * 2.0
-                self.pos.x += self.radius
-                is_pad_hit = True
-
-        if self.pos.x >= cpuPlayer.pos.x - cpuPlayer.rect_width:
-            if self.line_intersection(self.pos_before.x, self.pos_before.y, self.pos.x, self.pos.y, cpuPlayer.pos.x - cpuPlayer.rect_width / 2, cpuPlayer.pos.y - cpuPlayer.rect_height / 2, cpuPlayer.pos.x - cpuPlayer.rect_width / 2, cpuPlayer.pos.y + cpuPlayer.rect_height / 2):
-                self.pos.x = min(self.SCREEN_WIDTH, self.pos.x)
-                self.vel.x = -1 * (self.vel.x + self.speed * 0.05)
-                self.vel.y += cpuPlayer.vel.y * 0.006
-                self.pos.x -= self.radius
-                is_pad_hit = True
+        #Using pygame collision detection, we can pass in the objects directly, first the player
+        if pygame.sprite.collide_rect(self, agentPlayer):
+            self.pos.x = max(0, self.pos.x)
+            self.vel.x = -1 * (self.vel.x + self.speed * 0.05)
+            self.vel.y += agentPlayer.vel.y * 2.0
+            self.pos.x += self.radius
+            is_pad_hit = True
+
+        #then we test the cpu, much cleaner!
+        if pygame.sprite.collide_rect(self, cpuPlayer):
```

Comment:
Can you verify that this does not slow down the step speed? I believe using this function was slower.